project/apps/web/app/questions/components/CreateModal.tsx

```diff
@@ -0,0 +1,201 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { useZodForm } from "@/lib/form";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { CreateQuestionDto, createQuestionSchema } from "@repo/dtos/questions";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { Badge } from "@/components/ui/badge";
+import { CATEGORY, COMPLEXITY } from "@/constants/question";
+
+interface CreateModalProps {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  onCreate: (data: CreateQuestionDto) => void;
+}
+
+export default function CreateModal({
+  open,
+  setOpen,
+  onCreate,
+}: CreateModalProps) {
+  const form = useZodForm({
+    schema: createQuestionSchema,
+    defaultValues: {
+      q_title: "",
+      q_desc: "",
+      q_complexity: undefined,
+      q_category: [],
+    },
+  });
+
+  const categories = Object.values(CATEGORY);
+  const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
+
+  const toggleCategory = (category: string) => {
+    const updatedCategories = selectedCategories.includes(category)
+      ? selectedCategories.filter((cat) => cat !== category)
+      : [...selectedCategories, category];
+
+    setSelectedCategories(updatedCategories);
+    form.setValue("q_category", updatedCategories);
+  };
+
+  const handleSubmit = (data: CreateQuestionDto) => {
+    data.q_category = selectedCategories;
+    onCreate(data);
+    setOpen(false);
```

Comment:
should we remove this line? seems like `onCreate` already does `setOpen(false)`, this way we can also not close the modal if any errors happen and show it as a toast. 
```suggestion
  const handleSubmit = (data: CreateQuestionDto) => {
    onCreate(data);
```